soc/xtensa/espressif_esp32/esp32s3/default.ld

```diff
@@ -10,6 +10,8 @@
  * Linker script for the ESP32S3 platform.
  */
 
+#define Z_VM_KERNEL 1
```

Comment:
We have redefined the macro `GROUP_ROM_LINK_IN` at the beginning of the linker script to use our VMA `AT` LMA.

Reply:
Hi, there is a discussion about all of  this in the linked issue, the last comment is a summary.
But TLDR, for me including zbus and mcuboot breaks the alignment of the rodata segment and the c3 and s3 segfault on boot. The most general way to fix it was this. The definition of Z_VM_KERNEL makes the included scripts use ALIGN_WITH_INPUT for their sections which keeps the segment continuous. One side effect is that undefines GROUP_LINK_IN, which really just d I d what is explicitly done here anyway.
This fixed the segfaults for the test case on C3 and S3, another user had success on s2. There are other issues linked to this as well I suspect.

Reply:
Hi @dean-at-evos, I can understand that it "fixed your problems", but in the end, the changes will be discarded after the new hal based on v5.1 is merged. For reference please take a look at the C3 linker [here](https://github.com/marekmatej/zephyr/blob/feature/simple_boot_hal_v5.1/soc/riscv/espressif_esp32/esp32c3/default.ld). I have tested on my halv5.1 branches and the original issue is not there.

Reply:
That's great. It didn't "just fix my issue" it fixed a clear bug in the linking process. Evidence is a broken sample on two SOC and an application on two more. This change that brings your linker script much closer to the intent of the Z boiler plate, I could not break ... the original breaks by adding a log print.
So ... for a week and a half I was chasing this issue down, as it is a show stopper for a non hobby project. No one commented then, try the upcoming 5.1.... or frankly took it that seriously. I am not too concerned about how you resolve what it clearly a bug in these provided linker scripts, I also don't claim to be a ld expert. I thought to contribute and help others out ...

Reply:
@dean-at-evos, I read the discussions about the linker alignment issue. We have not spotted that issue before and your effort on finding and providing a solution is really appreciated.
We have been updating current hal_espressif (based on esp-idf v4.4) to use esp-idf v5.1 as baseline (and bring C6, etc.. support to Zephyr). This adds a bit of changes, including linker files. As Marek reported, this was fixed in there by his tests. However, this (new hal) changes will not be within this release v3.6 and I consider we do need to fix this during feature freeze window. Let us evaluate the proposed solution and contribute somehow.

Reply:
Thanks I probably went a little bit too hard on my previous comment and I apologise if I caused any offence.

I do appreciate your help in resolving issues as they arrive. I did get a bit upset  at the suggestion I am just selfishly pushing my own barrow. In fact my employer would probably prefer I did, I often justify time spent doing small maintenance and investigation as time spent for everyone's benefit. Even if our problem can be solved more rapidly.

Anyway, I'll look into what is proposed in 5.1 .. maybe that is a good merge. Although what I have for now keeps me moving on, we have a beta deployment at the end of this month.